specification/cognitiveservices/OpenAI.Inference/models/completions/extensions/oyd/mongodb_options.tsp

```diff
@@ -0,0 +1,64 @@
+import "@typespec/versioning";
+import "@azure-tools/typespec-client-generator-core";
+import "./../../azure_chat_extensions.tsp";
+import "./common_options.tsp";
+
+namespace Azure.OpenAI;
+
+using TypeSpec.Versioning;
+using Azure.ClientGenerator.Core;
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "MongoDB is a branded name"
+@added(ServiceApiVersions.v2024_08_01_Preview)
+@doc("""
+  A specific representation of configurable options for a MongoDB chat extension configuration.
+  """)
+model MongoDBChatExtensionConfiguration
+  extends AzureChatExtensionConfiguration {
+  @doc("""
+    The type label for MongoDB.
+    """)
+  type: AzureChatExtensionType.mongoDB;
+
+  @doc("The parameters for the MongoDB chat extension.")
+  parameters: MongoDBChatExtensionParameters;
+}
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "MongoDB is a branded name"
+@added(ServiceApiVersions.v2024_08_01_Preview)
+@doc("Parameters for the MongoDB chat extension. The supported authentication types are AccessToken, SystemAssignedManagedIdentity and UserAssignedManagedIdentity.")
+model MongoDBChatExtensionParameters {
+  ...OnYourDataCommonChatExtensionConfiguration;
```

Comment:
I think since we are applying a type here that is a super set of the options supported, this might be imprecise but still correct. Would love a 2nd opinion on this.

Reply:
We should not include auth schemes that are not supported by the specific data source/extension.

Reply:
We should not include auth schemes that are not supported by the specific data source/extension.

Reply:
Addressed by https://github.com/Azure/azure-rest-api-specs/pull/30377/files/8f988741692db5b8f414f5844131645092d4000a..d832bf585df96fef4b208357a7223b575ddbbf31